python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -115,9 +126,59 @@ def get_nodes_to_launch(
         total_nodes_to_add = self._get_concurrent_resource_demand_to_launch(
             total_nodes_to_add, usage_by_ip.keys(), nodes, pending_nodes)
 
+        if self.is_legacy_yaml:
+            assert len(total_nodes_to_add.keys()) <= 1
         logger.info("Node requests: {}".format(total_nodes_to_add))
         return total_nodes_to_add
 
+    def _handle_legacy_yaml(self, nodes: List[NodeID],
```

Comment:
TODO: launch only if worker and head node are different types.

Reply:
consider filling the cluster_config in prepare config as follows:
`config.avalable_node_types[TAG_RAY_NODE_TYPE_LEGACY] = config["worker_nodes"]`